src/ZkFold/Base/Algebra/Polynomials/Multivariate/Polynomial.hs

```diff
@@ -38,15 +39,20 @@ instance Polynomial c i j => FromPolynomial c i j m [(c, M i j m)] where
     fromPolynomial = id
 
 class Polynomial c i j => ToPolynomial c i j m p where
-    toPolynomial   :: [(c, M i j m)] -> Maybe p
+    toPolynomial   :: [(c, M i j m)] -> p
 
 instance (Polynomial c i j) => ToPolynomial c i j m [(c, M i j m)] where
-    toPolynomial   = Just . filter (\(c, _) -> c /= zero)
+    toPolynomial   = filter (\(c, _) -> c /= zero)
 
 -- | Polynomial type
 newtype P c i j m p = P p
     deriving (Generic, NFData, FromJSON, ToJSON)
 
+instance IsList (P c i j (Map i j) [(c, M i j (Map i j))]) where
```

Comment:
I think we can make it even more convenient by generalizing this to arbitrary `FromPolynomial c i j m p, ToPolynomial c i j m p`

Reply:
Those classes are already removed